src/lib/update-lock.ts

```diff
@@ -94,28 +122,56 @@ export const readLock: LockFn = Bluebird.promisify(locker.async.readLock, {
 function dispose(release: () => void): Bluebird<void> {
 	return Bluebird.map(_.keys(locksTaken), (lockName) => {
 		delete locksTaken[lockName];
-		return lockFile.unlockAsync(lockName);
+		return lockfile.unlock(lockName);
 	})
 		.finally(release)
 		.return();
 }
 
-const lockExistsErrHandler = (err: Error, release: () => void) => {
-	let errMsg = err.message;
-	if (EEXIST(err)) {
+const lockfileErrHandler = (err: Error, release: () => void) => {
+	let error: any;
+
+	// If error code is 73, updates.lock file already exists.
+	if (LockfileExistsError(err as ChildProcessError)) {
+		let errMsg = (err as ChildProcessError).stderr;
+
 		// Extract appId|appUuid and serviceName from lockfile path for log message
-		// appId: [0-9]{7}, appUuid: [0-9a-w]{32}, short appUuid: [0-9a-w]{7}
-		const pathMatch = err.message.match(
-			/\/([0-9]{7}|[0-9a-w]{32}|[0-9a-w]{7})\/(.*)\/(?:resin-)?updates.lock/,
+		// appId: [0-9]{7}, appUuid: [0-9a-f]{32}, short appUuid: [0-9a-f]{7}
+		const pathMatch = errMsg.match(
+			/\/([0-9]{7}|[0-9a-f]{32}|[0-9a-f]{7})\/(.*)\/(?:resin-)?updates.lock/,
```

Comment:
I made this silly mistake before and added 1204c40bbd21595cc18c124dab4daa9068338c1f to rectify.. UUID regex should be `[0-9a-f]` and I don't know why I wrote it as `[0-9a-w]`

Reply:
Fwiw I don't think is such a good idea to be parsing the error. The locking function can tell which one of the locking operations failed, and can return an `UpdatesLockedError` with the relevant info.

Reply:
No longer parsing the error 💯